src/main/java/at/hannibal2/skyhanni/features/mining/glacitemineshaft/CorpseTracker.kt

```diff
@@ -136,6 +140,20 @@ object CorpseTracker {
         }
     }
 
+    @SubscribeEvent
+    fun onItemAdd(event: ItemAddEvent) {
+        if (!isEnabled()) return
+        ProfileStorageData.profileSpecific?.mining?.mineshaft?.corpseProfitTracker?.let { trackerData ->
+            trackerData.getSelectedBucket()?.let {
+                tracker.addItem(it, event.internalName, event.amount)
+                ChatUtils.chat("Added ${event.internalName.itemName} §8x${event.amount} to Corpse Tracker §8(${it.displayName}§8)§e.")
```

Comment:
this should be abstracted into the bucketed item tracker.
ideally the bucketed item tracker would reuse the logic from skyhann iitem tracker so that this unique chat message is not necessary in the first place.

Reply:
We can't re-use the logic - they're entirely separate trackers. The item trackers just have the concept of one set of global "loot" that's added to - this is an enum driven base of buckets, the logic doesn't fit. The only way I could think to do this was using the currently-selected bucket.

Best I could do would be to abstract it back to the bucketed item tracker, however that message is going to end up having to be _very_ generic. We can't do it in `SkyhanniItemTracker` and expect it to work here - the `addItem` signatures are already overridden for `SkyhanniBucketedItemTracker`, as it takes a bucket as first param.

We also don't have the context of the `tracker` instance in the tracker itself, which would prevent actually moving this into the generic `SkyhanniBucketedItemTracker`, as we need to know the currently selected bucket;

```kotlin
ProfileStorageData.profileSpecific?.mining?.mineshaft?.corpseProfitTracker?.let {
}
```

Reply:
The only way I could think to do this was using the currently-selected bucket.

I was not expecting anything more.

Reply:
How about abstracting the existing item tracker class and creating a new "single bucketed tracker" for everything else?